fbzmq/async/AsyncSignalHandler.h

```diff
@@ -92,6 +98,10 @@ class AsyncSignalHandler {
 
   // registered signals
   sigset_t registeredSignals_;
+
+#ifdef IS_BSD
+  std::unordered_map<int, sig_t> signalHandlers_;
```

Comment:
Use array instead of map. There are limited number of signals & signal.h should have MAX signal value as well for bounded array. Most entries would go unused but performance benefit would be huge. 

Map lookup on every message would cost a heavy penalty.